addons/website_crm_partner_assign/models/res_partner.py

```diff
@@ -94,6 +94,16 @@ def _compute_opportunity_count(self):
             partner.opportunity_count += assign_counts.get(partner.id, 0)
 
     def action_view_opportunity(self):
+        self.ensure_one()  # especially here as we are doing an id, in, IDS domain
         action = super().action_view_opportunity()
-        action['domain'] = expression.OR([action.get('domain', []), [('partner_assigned_id', '=', self.id)]])
+        action_domain_origin = action.get('domain')
+        action_domain_assign = [('partner_assigned_id', '=', self.id)]
+        if not action_domain_origin:
+            action['domain'] = action_domain_assign
+            return action
+        # perform searches independently as having OR with those leaves seems to
+        # be counter productive
+        ids_origin = self.env['crm.lead'].search(action_domain_origin).ids
+        ids_new = self.env['crm.lead'].search(action_domain_assign).ids
```

Comment:
Those search change the behavior of the action. Only active opp are returned with this PR. The previous behavior was searching with active_test=False define in the context return by action_view_opportunity 

So to be consistent with the number shows on the partner page you should probably do this

```suggestion
        Lead = self.env['crm.lead'].with_context(active_test=False)
        ids_origin = Lead.search(action_domain_origin).ids
        ids_new = Lead.search(action_domain_assign).ids
```

Reply:
Good remark. I actually used the original context of super() to be inlined with possible customization people did.